core/common/src/RpcInterface.ts

```diff
@@ -34,20 +32,57 @@ export interface RpcInterfaceDefinition<T extends RpcInterface = RpcInterface> {
  */
 export type RpcInterfaceImplementation<T extends RpcInterface = RpcInterface> = new () => T;
 
+interface SemverType {
+  major: number;
+  minor: number;
+  patch: number;
+  prerelease?: string;
+}
+
 /** An RPC interface is a set of operations exposed by a service that a client can call, using configurable protocols,
  * in a platform-independent way. TheRpcInterface class is the base class for RPC interface definitions and implementations.
  * @public
  */
 export abstract class RpcInterface {
+
+  private static findDiff(backend: SemverType, frontend: SemverType): string {
+    const isSame = backend.major === frontend.major && backend.minor === frontend.minor && backend.patch === frontend.patch && backend.prerelease === frontend.prerelease;
+    if (isSame) {
```

Comment:
remove : string so this will return the correct type (the set of possible strings);

I don't see why you have the `isSame` test. Why not just:

```ts
private static findDiff(backend: SemverType, frontend: SemverType) {
return 
  backend.major !== frontend.major ? "major" : 
  backend.minor !== frontend.minor ? "minor" : 
  backend.patch !== frontend.patch ? "patch" : 
  backend.prerelease !== frontend.preprelase ? "prerelease" : 
  "same";
}
```

Reply:
modified function to that